spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc

```diff
@@ -885,6 +885,83 @@ would have all the data from those three sources sent to the file sink.
 The Fan-out use case is when you determine the destination of a stream based on some information that is only known at runtime.
 In this case, the link:http://docs.spring.io/spring-cloud-stream-app-starters/docs/Bacon.RELEASE/reference/html/spring-cloud-stream-modules-sinks.html#spring-cloud-stream-modules-router-sink[Router Application] can be used to specify how to direct the incoming message to one of N named destinations.
 
+[[spring-cloud-dataflow-stream-java-dsl]]
+== Stream Java DSL
+
+If you prefer to deploy streams programmatically, you can do so by using the fluent style DSL api provided by the `spring-cloud-dataflow-rest-client` module.
+The java DSL is a convenient wrapper around the `DataFlowTemplate` class that makes it simpler to create and deploy applications.
+Start by adding it to your project:
+
+[source,xml,subs="attributes+"]
+----
+<dependency>
+	<groupId>org.springframework.cloud</groupId>
+	<artifactId>spring-cloud-dataflow-rest-client</artifactId>
+	<version>{project-version}</version>
+</dependency>
+----
+
+To create a Stream you need to supply a `DataFlowTemplate` to the `Stream.Builder` static method:
+[source,java]
+----
+DataFlowTemplate client = //...;
+Stream ticktock = Stream.Builder(client).name("ticktock").definition("time | log").create();
+----
+[NOTE]
+For your convenience, the `spring-cloud-dataflow-rest-client` library provides a bean of type `StreamBuilder` with the `DataFlowTemplate` already configured.
+
+=== Creating Streams using the Java DSL
+
+The java DSL offers two styles to create Streams, if you prefer the pipes and filters DSL provided by dataflow you can use the `definition()` method and provide your own DSL definition.
+If you prefer a more fluent style usage, you can daisy chain methods via the `source()`, `processor()`, `sink()` methods.
+To demonstrate both styles we will create a simple stream using both approaches.
```

Comment:
Need to show that one needs to have the applications in the stream already registered.  This can be 'out of band' or showing how to do this with dataFlowOperations makes the docs more self contained.